sdk-base/src/main/java/com/mapbox/maps/plugin/viewport/experimental/ViewportStatus.kt

```diff
@@ -0,0 +1,60 @@
+package com.mapbox.maps.plugin.viewport.experimental
+
+import com.mapbox.maps.plugin.viewport.experimental.state.ViewportState
+import com.mapbox.maps.plugin.viewport.experimental.transition.ViewportTransition
+
+/**
+ * Represents the status of the viewport.
+ *
+ * It could be either a [ViewportState] or [ViewportTransition].
+ */
+sealed class ViewportStatus {
+  /**
+   * Represents the current status is a [ViewportState].
+   *
+   * The state is null if current status is IDLE.
+   */
+  class State(
+    /**
+     * The current [ViewportState].
+     */
+    val state: ViewportState?
+  ) : ViewportStatus() {
+    /**
+     * Get the next [ViewportState] regardless whether it's a transition or state.
+     */
+    override fun getTargetState(): ViewportState? {
```

Comment:
Feels like a a bit of a hack to me, why is it needed?

Reply:
Moreover users have access to public `ViewportStatus.toState` themselves

Reply:
yes, also talked with @macdrevx and agreed to remove it.